src/NServiceBus.TransactionalSession.AcceptanceTests.MongoDB/When_not_using_outbox.cs

```diff
@@ -0,0 +1,105 @@
+﻿namespace NServiceBus.TransactionalSession.AcceptanceTests
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using AcceptanceTesting;
+    using NUnit.Framework;
+
+    [ExecuteOnlyForEnvironmentWith(EnvironmentVariables.MongoDBConnectionString)]
+    public class When_not_using_outbox : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_send_messages_on_transactional_session_commit()
+        {
+            await Scenario.Define<Context>()
+                .WithEndpoint<AnEndpoint>(s => s.When(async (_, ctx) =>
+                {
+                    using var scope = ctx.ServiceProvider.CreateScope();
+                    using var transactionalSession = scope.ServiceProvider.GetRequiredService<ITransactionalSession>();
+
+                    await transactionalSession.OpenMongoDBSession();
+
+                    await transactionalSession.SendLocal(new SampleMessage());
+
+                    await transactionalSession.Commit();
+                }))
+                .Done(c => c.MessageReceived)
+                .Run();
+        }
+
+        [Test]
+        public async Task Should_not_send_messages_if_session_is_not_committed()
+        {
+            var result = await Scenario.Define<Context>()
+                .WithEndpoint<AnEndpoint>(s => s.When(async (messageSession, ctx) =>
+                {
+                    using (var scope = ctx.ServiceProvider.CreateScope())
+                    using (var transactionalSession = scope.ServiceProvider.GetRequiredService<ITransactionalSession>())
+                    {
+                        await transactionalSession.OpenMongoDBSession();
+
+                        await transactionalSession.SendLocal(new SampleMessage());
+                    }
+
+                    //Send immediately dispatched message to finish the test
+                    await messageSession.SendLocal(new CompleteTestMessage());
+                }))
+                .Done(c => c.CompleteMessageReceived)
+                .Run()
+                ;
```

Comment:
Nit:

```suggestion
                .Run();
```

but we can also leave this out. Not a big deal